core/arch/interp.c

```diff
@@ -817,6 +817,26 @@ check_new_page_jmp(dcontext_t *dcontext, build_bb_t *bb, app_pc new_pc)
 }
 
 static inline void
+bb_process_single_step(dcontext_t *dcontext, build_bb_t *bb)
+{
+    LOG(THREAD, LOG_INTERP, 2, "interp: single step exception bb at "PFX"\n", bb->instr_start);
+    instrlist_append(bb->ilist, bb->instr);
```

Comment:
Qin came up with yet another complication: if this is a rep stringop, presumably the real single step will execute just one iteration, while we (with no client using drutil_expand_rep_string()) will run the entire loop.  One hack would be to mangle it to store the real count in xcx somewhere, set it to 1 (unless already 0), and restore afterward.

Seems fine to put a FIXME for now, add a note to the issue tracker, and have that be a separate CL :)